### PR TITLE
Fix the bug that getImplicitUsersForPermission() cannot get users in role, similar to Go PR: https://github.com/casbin/casbin/pull/443

### PR DESCRIPTION
**Pull request by zhhzone**
Fix: https://github.com/php-casbin/php-casbin/issues/65

### CODE REVIEW

**Comment by hsluoyz** [comment]
@zhhzone modify your commit message to this:

```
Fix the bug that getImplicitUsersForPermission() cannot get users in role, similar to Go issue #443
```

**Comment by zhhzone** [comment]
@hsluoyz  It's done

**Comment by hsluoyz** [comment]
@techone please review.

**Comment by techoner** [comment]
@zhhzone Is there any change compared to the last commit ?

File: tests/Unit/EnforcerTest.php [hunk 1]
```diff
@@ -241,9 +241,9 @@ public function testGetImplicitPermissionsForUser()
...
     public function testGetImplicitUsersForPermission()
     {
-        $e = new Enforcer($this->modelAndPolicyPath . '/rbac_model.conf', $this->modelAndPolicyPath . '/rbac_with_hierarchy_policy.csv');
+        $e = new Enforcer($this->modelAndPolicyPath . '/rbac_model.conf', $this->modelAndPolicyPath . '/rbac_policy.csv');
+
         $this->assertEquals($e->getImplicitUsersForPermission('data1', 'read'), ['alice']);
-        $this->assertEquals($e->getImplicitUsersForPermission('data1', 'write'), ['alice']);
         $this->assertEquals($e->getImplicitUsersForPermission('data2', 'read'), ['alice']);
         $this->assertEquals($e->getImplicitUsersForPermission('data2', 'write'), ['alice', 'bob']);
     }
```
[the remark this answers is not in the document]
Comment (techoner) [reply tests/Unit/EnforcerTest.php:249]: I think this test case is not the same as the following(in go):
```
func testGetImplicitUsers(t *testing.T, e *Enforcer, res []string, permission ...string) {
	t.Helper()
	myRes, _ := e.GetImplicitUsersForPermission(permission...)
	t.Log("Implicit users for permission: ", permission, ": ", myRes)

	sort.Strings(res)
	sort.Strings(myRes)

	if !util.ArrayEquals(res, myRes) {
		t.Error("Implicit users for permission: ", permission, ": ", myRes, ", supposed to be ", res)
	}
}

func TestImplicitUserAPI(t *testing.T) {
	e, _ := NewEnforcer("examples/rbac_model.conf", "examples/rbac_with_hierarchy_policy.csv")

	testGetImplicitUsers(t, e, []string{"alice"}, "data1", "read")
	testGetImplicitUsers(t, e, []string{"alice"}, "data1", "write")
	testGetImplicitUsers(t, e, []string{"alice"}, "data2", "read")
	testGetImplicitUsers(t, e, []string{"alice", "bob"}, "data2", "write")

	e.ClearPolicy()
	e.AddPolicy("admin", "data1", "read")
	e.AddPolicy("bob", "data1", "read")
	e.AddGroupingPolicy("alice", "admin")
	testGetImplicitUsers(t, e, []string{"alice", "bob"}, "data1", "read")
}
```

File: tests/Unit/EnforcerTest.php [hunk 2]
```diff
@@ -241,9 +241,9 @@ public function testGetImplicitPermissionsForUser()
 
     public function testGetImplicitUsersForPermission()
     {
-        $e = new Enforcer($this->modelAndPolicyPath . '/rbac_model.conf', $this->modelAndPolicyPath . '/rbac_with_hierarchy_policy.csv');
+        $e = new Enforcer($this->modelAndPolicyPath . '/rbac_model.conf', $this->modelAndPolicyPath . '/rbac_policy.csv');
+
```
Comment (techoner) [inline tests/Unit/EnforcerTest.php:245]: Why modify the policy file ?

File: tests/Unit/EnforcerTest.php [hunk 3]
```diff
@@ -241,9 +241,9 @@ public function testGetImplicitPermissionsForUser()
...
         $this->assertEquals($e->getImplicitUsersForPermission('data1', 'read'), ['alice']);
-        $this->assertEquals($e->getImplicitUsersForPermission('data1', 'write'), ['alice']);
```
Comment (techoner) [inline tests/Unit/EnforcerTest.php:246]: Why ?

**Review by techoner — CHANGES_REQUESTED** [review]

File: tests/Unit/EnforcerTest.php (hunk 2 above)
Comment (zhhzone) [reply tests/Unit/EnforcerTest.php:245]: This policy should use the old one here.   My fault

File: tests/Unit/EnforcerTest.php (hunk 3 above)
Comment (zhhzone) [reply tests/Unit/EnforcerTest.php:246]:
```php
   $this->assertEquals($e->getImplicitUsersForPermission('data1', 'read'), ['alice']);
   $this->assertEquals($e->getImplicitUsersForPermission('data1', 'write'), ['alice']);

```
I think these two pieces of code have the same effect. Although they used different variables

**Comment by hsluoyz** [comment]
@techoner

**Comment by techoner** [comment]
@zhhzone Please refer to the test case of Casbin(go).

**Pull request closed without merging**